### ISSUE TITLE
[Feature Request] Add integration with validation libraries (eg `zod`)

### ISSUE THREAD

**Issue by Lonli-Lokli** [cyclosproject/ng-openapi-gen#270]
It's mostly a discussion, not a feature request, as I still don't know is it possible with template generating.

There are multiple libraries for validating responses from API; I prefer to use [zod](https://github.com/colinhacks/zod).
And I would like to have an angular HTTP client with data validation automatically, while now I have to do this
1) Generate default client from openapi schema with [ng-openapi-gen](https://github.com/cyclosproject/ng-openapi-gen)
2) Generate zod schema from openapi with https://github.com/josejulio/openapi2typescript/tree/main/packages/cli
3) Create wrappers for each service to put zod schemas

Basically, I would like to have smth similar to https://github.com/ecyrbe/zodios-openapi but with Angular

**Comment by dsinghvi** [comment]
@Lonli-Lokli I work on an open source project called [Fern](https://github.com/fern-api/fern) and our generated TypeScript client comes with request/response validation built in. Our clients have a serde layer where we validate the JSON and process the input in certain cases (e.g. turning string into `Date`, turning an array into a `Set`). Would that be useful?

**Comment by luisfpg** [comment]
Sorry for late reply.
I consider ng-openapi-gen feature-complete, and I have very short time for it nowadays.
So the idea is to only include bug fixes...
